lib/mergeArrayByName.js

```diff
@@ -0,0 +1,34 @@
+// https://github.com/KyleAMathews/deepmerge#arraymerge
+
+const merge = require('deepmerge')
+
+function findMatchingIndex (sourceItem, target) {
+  var matchingIndex
+  if (Object.keys(sourceItem).includes('name')) {
+    matchingIndex = target
+      .filter((targetItem) => {
```

Comment:
Sorry for the nitpick, but I think these would be best as one-liners:

```suggestion
      .filter(targetItem => targetItem.hasOwnProperty('name'))
```